File: src/main/scala/assignment1/Group.scala
package assignment1

class Group[T]:
  private var contents = List[T]()

  def add(x: T): Unit = {
    if !has(x)
    then contents = contents :+ x;
  }

  def delete(x: T): Unit = {
    try {
      if has(x)
      then {
        contents = contents.filterNot(_.equals(x))
      } else {
        throw new Exception("There is no such an element")
      }
    } catch {
      case ex: Exception => ;
    }
  }

  def has(x: T): Boolean = {
    if (contents.contains(x))
    then {
      true
    } else { false }
  }

object Group:
  def from[T](xs: Iterable[T]): Group[T] = {
    var result = Group[T]
    for x <- xs do {
      result.add(x)
    }
    result
  }

@main def test() = {
  val group = Group.from(List(10, 20))
  println(group.has(10))
  // → true
  println(group.has(30))
  // → false
  group.add(10)
  group.delete(10)
  println(group.has(10))
  // → false
}

File: src/main/js/assignment1/Group.js
class Group {
    #contents;

    constructor() {
        this.#contents = [];
    }

    has(x) {
        return this.#contents.includes(x);
    }

    add(x) {
        if (!this.has(x)) {
            this.#contents.push(x);
        }
    }

    delete(x) {
        try{
            if (this.has(x)){
                this.#contents = this.#contents.filter(element => element !== x);
            }else{
                throw new Error('There is no such an element')
            }
        }catch{
            ;
        }
    }

    static from(it) {
        let result = new Group();
        for (let x of it) {
            result.add(x);
        }
        return result;
    }
}

function main() {
    let group = Group.from([10, 20]);
    console.log(group.has(10));
    // → true
    console.log(group.has(30));
    // → false
    group.add(10);
    group.delete(10);
    console.log(group.has(10));
    // → false
}

if (require.main === module) {
    main();
}

module.exports = Group;